apps/admin-gui/src/app/facilities/pages/facility-detail-page/facility-settings/facility-settings-blacklist/facility-settings-blacklist.component.ts

```diff
@@ -0,0 +1,75 @@
+import { Component, OnInit } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import {
+  BanOnFacility,
+  FacilitiesManagerService,
+  Facility,
+  User,
+  UsersManagerService
+} from '@perun-web-apps/perun/openapi';
+import { TABLE_FACILITY_BLACKLIST_LIST, TableConfigService } from '@perun-web-apps/config/table-config';
+import { ActivatedRoute } from '@angular/router';
+import { SelectionModel } from '@angular/cdk/collections';
+import { PageEvent } from '@angular/material/paginator';
+
+@Component({
+  selector: 'app-perun-web-apps-facility-settings-blacklist',
+  templateUrl: './facility-settings-blacklist.component.html',
+  styleUrls: ['./facility-settings-blacklist.component.scss']
+})
+export class FacilitySettingsBlacklistComponent implements OnInit {
+
+  constructor(private dialog: MatDialog,
+              private facilitiesManager: FacilitiesManagerService,
+              private usersManager: UsersManagerService,
+              private tableConfigService: TableConfigService,
+              private route: ActivatedRoute) {
+  }
+
+  facility: Facility;
+  bansOnFacilitiesWithUsers: [BanOnFacility, User][] = [];
+  selected = new SelectionModel<[BanOnFacility, User]>(true, []);
+  filterValue = '';
+  loading: boolean;
+  pageSize: number;
+  tableId = TABLE_FACILITY_BLACKLIST_LIST;
+
+
+  ngOnInit(): void {
+    this.pageSize = this.tableConfigService.getTablePageSize(this.tableId);
+
+    this.route.parent.parent.params.subscribe(parentParentParams => {
+      const facilityId = parentParentParams['facilityId'];
+      this.facilitiesManager.getFacilityById(facilityId).subscribe(facility => {
```

Comment:
If I am not mistaken you only need for whole component only FacilityId, so it is for nothing to call this function here right?